backend/src/web-server/controllers/history-query.controller.ts

```diff
@@ -439,4 +442,98 @@ export default class HistoryQueryController extends AbstractController {
     }
     ctx.noContent();
   }
+
+  async testSouthConnection(ctx: KoaContext<SouthConnectorCommandDTO, void>) {
+    try {
+      const manifest = ctx.request.body
+        ? ctx.app.southService.getInstalledSouthManifests().find(southManifest => southManifest.id === ctx.request.body!.type)
+        : null;
+      if (!manifest) {
+        return ctx.throw(404, 'South manifest not found');
+      }
+
+      let southSettings: SouthConnectorDTO['settings'] | null = null;
+      // When we have a history query id, we can retrieve the old connector settings from repository
+      if (ctx.params.id !== 'create') {
+        southSettings = ctx.app.repositoryService.historyQueryRepository.getHistoryQuery(ctx.params.id)?.southSettings;
+        if (!southSettings) {
+          return ctx.notFound();
+        }
+      }
+      // When creating a new history query from exising connector, we can retrieve old connector settings using connector id
+      else if (ctx.request.query?.fromConnectorId && typeof ctx.request.query?.fromConnectorId === 'string') {
```

Comment:
I've updated the tests but I couldn't get them to 100%.

At this `else if` (both here and for north test) seems to be the issue. <br>
I have tests that check cases where `fromConnectorId` isn't set, and where it's not a string. I think the issue is with the string checking.

Could you take a look as well? Or do you have any suggestions regarding this?